### PR TITLE
Document formatting patterns used in Time formatting

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
This is to document formatting patterns used in Time formatting.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/3982

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 1]
```diff
@@ -99,3 +99,51 @@ time:Time time3 = time:currentTime();
...
+''|single quote|literal|'
+[|optional section start|
+]|optional section end|
+
+
+| Tables        | Are           | Cool  |
+| ------------- |:-------------:| -----:|
+| col 3 is      | right-aligned | $1600 |
+| col 2 is      | centered      |   $12 |
+| zebra stripes | are neat      |    $1 |
```
Comment (anupama-pathirage) [inline stdlib/time/src/main/ballerina/src/time/Module.md]: What is this table?

Comment (praneesha) [reply stdlib/time/src/main/ballerina/src/time/Module.md]: Removed it.

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 2]
```diff
@@ -99,3 +99,45 @@ time:Time time3 = time:currentTime();
 time:Time time4 = time:currentTime();
 int diffInMillis = time3.time - time4.time;
 ```
+## Patterns for formatting and parsing
+
+The below patterns can be used for `TimeFormat` when using the `format()` and `parse()` functions.
```
Comment (anupama-pathirage) [inline stdlib/time/src/main/ballerina/src/time/Module.md]:
```suggestion
The below patterns can be used to generate the formatter string when using the `format()` and `parse()` functions.
```

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 3]
```diff
@@ -99,3 +99,45 @@ time:Time time3 = time:currentTime();
...
+
+**Symbol**|**Meaning**|**Presentation**|**Examples**
+:-----:|:-----:|:-----:|:-----:
```
Comment (anupama-pathirage) [inline stdlib/time/src/main/ballerina/src/time/Module.md]: Why do we need dashed lines here?

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 4]
```diff
@@ -99,3 +99,45 @@ time:Time time3 = time:currentTime();
...
+z|time-zone name|zone-name|Pacific Standard Time; PST
+O|localized zone-offset|offset-O|GMT+8; GMT+08:00; UTC-08:00;
+X|zone-offset 'Z' for zero|offset-X|Z; -08; -0830; -08:30; -083015; -08:30:15;
+x|zone-offset|offset-x|+0000; -08; -0830; -08:30; -083015; -08:30:15;
+p|pad next|pad modifier|1
+'|escape for text|delimiter|
+''|single quote|literal|'
+[|optional section start|
+]|optional section end|
+
```
Comment (anupama-pathirage) [inline stdlib/time/src/main/ballerina/src/time/Module.md]: Extra new line

File: stdlib/time/src/main/ballerina/src/time/Module.md (hunk 2 above)
Comment (praneesha) [reply stdlib/time/src/main/ballerina/src/time/Module.md]: Fixed.

File: stdlib/time/src/main/ballerina/src/time/Module.md (hunk 4 above)
Comment (praneesha) [reply stdlib/time/src/main/ballerina/src/time/Module.md]: Fixed.

File: stdlib/time/src/main/ballerina/src/time/Module.md (hunk 3 above)
Comment (praneesha) [reply stdlib/time/src/main/ballerina/src/time/Module.md]: Fixed.

**Pull request merged**